### PR TITLE
Support installing the ASP.NET Core shared runtime from KoreBuild

### PR DESCRIPTION
**Pull request by natemcmaster**
Required for templating, MusicStore, and others that will need to perform functional tests with the aspnetcore shared runtime

### CODE REVIEW

File: modules/KoreBuild.Tasks/Internal/DotNetAssetRequest.cs [hunk 1]
```diff
@@ -2,22 +2,68 @@
...
+            public AspNetCoreRuntime(string version) : base(version)
+            {
+            }
+
+            public override string RuntimeName { get; } = "aspnetcore";
+
+            public override string DisplayName { get; } = "ASP.NET Core Runtime";
+
+            public override bool IsInstalled(string basePath)
+                => File.Exists(Path.Combine(basePath, "shared", "Microsoft.AspNetCore.All", Version, ".version"));
```
Comment (natemcmaster) [inline modules/KoreBuild.Tasks/Internal/DotNetAssetRequest.cs]: @JunTaoLuo does this file exist in our shared framework?

Comment (JunTaoLuo) [reply modules/KoreBuild.Tasks/Internal/DotNetAssetRequest.cs]: Yup

File: files/KoreBuild/scripts/dotnet-install.ps1 [hunk 2]
```diff
@@ -71,6 +80,9 @@ param(
    [string]$Version="Latest",
    [string]$InstallDir="<auto>",
    [string]$Architecture="<auto>",
+   [ValidateSet("dotnet", "aspnetcore", IgnoreCase = $false)]
```
Comment (ryanbrandenburg) [inline files/KoreBuild/scripts/dotnet-install.ps1]: Neat! Poor-man's `Enum`.

**Comment by ryanbrandenburg** [comment]
Looks like the tests are [failing](https://travis-ci.org/aspnet/BuildTools/builds/317188199#L883).

**Comment by natemcmaster** [comment]
Yup, forgot to update tests. They should pass now

File: files/KoreBuild/scripts/dotnet-install.ps1 (hunk 2 above)
Comment (natemcmaster) [reply files/KoreBuild/scripts/dotnet-install.ps1]: Basically yeah, it's like an enum. Even cooler, PowerShell's tab completion will cycle through valid options only

**Comment by natemcmaster** [comment]
Yup. The guy who can grant access to the azure account we need to push to is OOF.

**Comment by JunTaoLuo** [comment]
@pranavkm you have push access to dotnetcli right?

**Comment by natemcmaster** [comment]
I was granted access yesterday. I’ll finish getting this setup after the holiday.

**Comment by javiercn** [comment]
Sounds good to me :shipit:

**Pull request merged**